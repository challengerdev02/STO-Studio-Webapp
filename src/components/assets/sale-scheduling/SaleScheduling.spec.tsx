import {
  cleanup,
  fireEvent,
  queryAllByAttribute,
  queryByAttribute,
  queryByPlaceholderText,
  queryByText,
  render,
  waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { SaleScheduling } from './index';

describe('SaleScheduling', function () {
  afterEach(cleanup);

  let props = {
    loading: true,
    title: 'Spider man: Home coming',
    selectedForm: 'Buy',
    onFormTypeChange: jest.fn(),
    onFinish: jest.fn(),
    onCancel: jest.fn(),
    children: <div data-testid={'modal-test'}>This is a test child</div>,
  };
  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<SaleScheduling {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should have form tag', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryByAttribute(
        'class',
        document.body,
        'ant-form ant-form-vertical ant-form-hide-required-mark'
      )
    ).toHaveAttribute('id', 'sale-scheduling-form');
  });
  it('should have form radio tag', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryAllByAttribute('type', document.body, 'radio')[0]
    ).toBeChecked();
  });
  it('should have title "Spider man: Home coming"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(
      queryByText(document.body, 'Spider man: Home coming')
    ).toBeInTheDocument();
  });

  it('should have title "copies available "', () => {
    // @ts-ignore
    const {} = setup();
    expect(queryByText(document.body, 'copies available')).toBeInTheDocument();
  });

  it('should have title "Price (ETHER)"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByText(document.body, 'Price (ETHER)')).toBeInTheDocument();
  });

  it('should have input "Enter number of copies"', () => {
    // @ts-ignore
    const {} = setup();
    expect(
      queryByPlaceholderText(document.body, 'Enter number of copies')
    ).toBeInTheDocument();
    expect(
      queryByPlaceholderText(document.body, 'Enter number of copies')
    ).toHaveClass('ant-input-number-input');
  });

  // it('should have input "You will be charged 2 ETH"', () => {
  //   // @ts-ignore
  //   const {} = setup();
  //   expect(
  //     queryByText(document.body, 'You will be charged 2 ETH')
  //   ).toBeInTheDocument();
  // });

  it('should have date picker ', () => {
    // @ts-ignore
    const {} = setup();
    expect(
      queryAllByAttribute('class', document.body, 'ant-picker')
    ).toHaveLength(1);
  });
  it('should have date picker panel', async () => {
    // @ts-ignore
    const {} = setup();
    await waitFor(() => {
      const input = queryAllByAttribute(
        'id',
        document.body,
        'sale-scheduling-form_startDate'
      )[0];

      if (input) {
        fireEvent.mouseDown(input);
      }
    });

    await waitFor(() => {
      expect(
        queryAllByAttribute('class', document.body, 'ant-picker')
      ).toHaveLength(1);
      expect(
        queryAllByAttribute(
          'class',
          document.body,
          'ant-picker-panel-container'
        )
      ).toHaveLength(1);
    });
  });

  it('should Not have Auto saving ', () => {
    // @ts-ignore
    const {} = setup({ loading: false });
    expect(queryByText(document.body, 'Auto saving')).not.toBeInTheDocument();
  });

  it('should have submit button ', () => {
    // @ts-ignore
    const {} = setup();
    expect(
      queryByAttribute('type', document.body, 'submit')
    ).toBeInTheDocument();
    expect(queryByAttribute('type', document.body, 'submit')).toContainHTML(
      `<span>Schedule Sale </span>`
    );
  });

  it('should show aution form', async () => {
    // @ts-ignore
    const { container } = setup({ selectedForm: 'Auction' });

    await waitFor(() => {
      const btn = queryByAttribute('value', document.body, 'Auction');

      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      const btn = queryByAttribute('value', document.body, 'Auction');
      expect(btn).toBeInTheDocument();
      expect(
        queryAllByAttribute('type', document.body, 'radio')[1]
      ).toBeChecked();
      expect(queryByText(document.body, 'duration')).toBeInTheDocument();
      expect(queryByAttribute('type', document.body, 'submit')).toContainHTML(
        `<span>Schedule Sale </span>`
      );
    });
  });

  it('should show Launch Event form', async () => {
    // @ts-ignore
    const { container } = setup({ selectedForm: 'Launch' });

    await waitFor(() => {
      const btn = queryByAttribute('value', document.body, 'Launch');

      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(
        queryAllByAttribute('type', document.body, 'radio')[2]
      ).toBeChecked();
      expect(queryByText(document.body, 'phone number')).toBeInTheDocument();
      expect(queryByAttribute('type', document.body, 'submit')).toContainHTML(
        `<span>Schedule Launch </span>`
      );
    });
  });
});
