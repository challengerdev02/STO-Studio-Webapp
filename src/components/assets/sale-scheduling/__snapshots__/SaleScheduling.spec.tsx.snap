// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaleScheduling should match with snapshot 1`] = `
<div>
  <div
    class="sc-gsDKAQ gDWIQW meta-form-container"
  >
    <span
      class="ant-typography sc-dkPtRN fHOetg"
    >
      Spider man: Home coming
    </span>
    <div
      class="ant-card ant-card-bordered sc-hKwDye bDDTra"
    >
      <div
        class="ant-card-body"
      >
        <form
          class="ant-form ant-form-vertical ant-form-hide-required-mark"
          id="sale-scheduling-form"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-radio-group ant-radio-group-outline"
                    id="sale-scheduling-form_salesType"
                  >
                    <label
                      class="ant-radio-wrapper"
                    >
                      <span
                        class="ant-radio"
                      >
                        <input
                          class="ant-radio-input"
                          type="radio"
                          value="BuyNow"
                        />
                        <span
                          class="ant-radio-inner"
                        />
                      </span>
                      <span>
                        Buy Now
                      </span>
                    </label>
                    <label
                      class="ant-radio-wrapper"
                    >
                      <span
                        class="ant-radio"
                      >
                        <input
                          class="ant-radio-input"
                          type="radio"
                          value="Auction"
                        />
                        <span
                          class="ant-radio-inner"
                        />
                      </span>
                      <span>
                        Auction
                      </span>
                    </label>
                    <label
                      class="ant-radio-wrapper"
                    >
                      <span
                        class="ant-radio"
                      >
                        <input
                          class="ant-radio-input"
                          type="radio"
                          value="Launch"
                        />
                        <span
                          class="ant-radio-inner"
                        />
                      </span>
                      <span>
                        Launch Event
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item submit-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn ant-btn-primary"
                    type="submit"
                  >
                    <span>
                       
                    </span>
                    <i
                      class="mc-arrow-right-2-line mc-lg"
                    />
                  </button>
                  <div
                    class="auto-container"
                  >
                     
                    <span
                      class="sc-bdvvtL cqkQrX"
                      role="1"
                    >
                      <svg
                        fill="none"
                        height="14"
                        viewBox="0 0 14 14"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7 0.333374C6.63181 0.333374 6.33333 0.631851 6.33333 1.00004V3.00004C6.33333 3.36823 6.63181 3.66671 7 3.66671C7.36819 3.66671 7.66667 3.36823 7.66667 3.00004V1.00004C7.66667 0.631851 7.36819 0.333374 7 0.333374Z"
                          fill="#FCFCFD"
                        />
                        <path
                          d="M7 10.3334C6.63181 10.3334 6.33333 10.6319 6.33333 11V13C6.33333 13.3682 6.63181 13.6667 7 13.6667C7.36819 13.6667 7.66667 13.3682 7.66667 13V11C7.66667 10.6319 7.36819 10.3334 7 10.3334Z"
                          fill="#FCFCFD"
                        />
                        <path
                          d="M13 6.33337C13.3682 6.33337 13.6667 6.63185 13.6667 7.00004C13.6667 7.36823 13.3682 7.66671 13 7.66671H11C10.6318 7.66671 10.3333 7.36823 10.3333 7.00004C10.3333 6.63185 10.6318 6.33337 11 6.33337H13Z"
                          fill="#FCFCFD"
                        />
                        <path
                          d="M3.66667 7.00004C3.66667 6.63185 3.36819 6.33337 3 6.33337H0.999999C0.631809 6.33337 0.333332 6.63185 0.333332 7.00004C0.333332 7.36823 0.631809 7.66671 0.999999 7.66671H3C3.36819 7.66671 3.66667 7.36823 3.66667 7.00004Z"
                          fill="#FCFCFD"
                        />
                        <path
                          d="M10.7713 2.28606C11.0316 2.02571 11.4537 2.02571 11.7141 2.28606C11.9744 2.54641 11.9744 2.96852 11.7141 3.22887L10.2999 4.64308C10.0395 4.90343 9.61741 4.90343 9.35706 4.64308C9.09671 4.38273 9.09671 3.96062 9.35706 3.70027L10.7713 2.28606Z"
                          fill="#FCFCFD"
                        />
                        <path
                          d="M4.64295 9.35703C4.3826 9.09668 3.96049 9.09668 3.70014 9.35703L2.28593 10.7712C2.02558 11.0316 2.02558 11.4537 2.28593 11.714C2.54628 11.9744 2.96839 11.9744 3.22874 11.714L4.64295 10.2998C4.9033 10.0395 4.9033 9.61737 4.64295 9.35703Z"
                          fill="#FCFCFD"
                        />
                        <path
                          d="M11.7141 10.7713C11.9744 11.0317 11.9744 11.4538 11.7141 11.7141C11.4537 11.9745 11.0316 11.9745 10.7713 11.7141L9.35704 10.2999C9.09669 10.0396 9.09669 9.61745 9.35704 9.3571C9.61739 9.09675 10.0395 9.09675 10.2998 9.3571L11.7141 10.7713Z"
                          fill="#FCFCFD"
                        />
                        <path
                          d="M4.64301 4.64299C4.90336 4.38265 4.90336 3.96054 4.64301 3.70019L3.2288 2.28597C2.96845 2.02562 2.54634 2.02562 2.28599 2.28597C2.02564 2.54632 2.02564 2.96843 2.28599 3.22878L3.70021 4.64299C3.96056 4.90334 4.38267 4.90334 4.64301 4.64299Z"
                          fill="#FCFCFD"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
